lms/djangoapps/courseware/models.py

```diff
@@ -230,3 +230,20 @@ class Meta:
 
     def __unicode__(self):
         return "[OCGLog] %s: %s" % (self.course_id.to_deprecated_string(), self.created)  # pylint: disable=no-member
+
+
+class StudentFieldOverride(models.Model):
+    """
+    Holds the value of a specific field overriden for a student.  This is used
+    by the code in the `courseware.student_field_overrides` module to provide
+    overrides of xblock fields on a per user basis.
+    """
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    location = LocationKeyField(max_length=255, db_index=True)
+    student = models.ForeignKey(User, db_index=True)
+
+    class Meta:   # pylint: disable=missing-docstring
+        unique_together = (('course_id', 'location', 'student'),)
+
+    field = models.CharField(max_length=255)
+    value = models.TextField(default='null')
```

Comment:
I think that this is a really interesting piece of functionality. Some questions/requests:

1. The unique constraint index is going to make (course_id, location) queries across students efficient, but I think that (course_id, student) queries across a user's state in a course are going to be more common. Maybe make the composite index on (student, course, location, field)?
2. Could you please add timestamps (inherit from model_utils.TimeStampedModel)?
3. We'll have to make sure that docs and analytics are aware that this will need to be part of the data export.
4. Should the unique constraint include the field (that maps to the xblock field name, right?)
5. If we don't have historical data for this, it may be harder to reconstruct a student's state across the run of the course (like we can do with `StudentModuleHistory`).

I would like to get an idea of where this is going in the longer term. Having such a general data model in place is very powerful, but could also potentially lead to a lot of hard to debug state if it starts getting used by different pieces of the system.